packages/flutter/lib/src/gestures/binding.dart

```diff
@@ -363,10 +372,25 @@ mixin GestureBinding on BindingBase implements HitTestable, HitTestDispatcher, H
     _handlePointerEventImmediately(event);
   }
 
+  /// Returns true if the [HitTestResult] given to [dispatchEvent] is a cached
+  /// one.
+  ///
+  /// Tapping (and pan zooming) uses the same hit test result throughout the
+  /// tap sequence, but hovering needs fresh hit tests every frame.
+  @protected
+  bool useCachedHitTestResult(PointerEvent event) {
```

Comment:
It took me a good amount of time to understand the purpose and semantics of `useCachedHitTestResult`. Among other things, it's kinda confusing that - as the comment below admits - it's not actually telling you about all caching instances. It seems to be very much tailored to the needs of the MouseTracker instead of being a general purpose API. Another reason for maybe considering to have the MouseTracker internally decide whether to reuse the hitTestResult from the pointer down (see my other comment on this).